### ISSUE TITLE
ORC input stream ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by usberkeley** [trinodb/trino#19934]
- System environment：Trino 433 + Hive + HDFS + ORC 
- Recurrence conditions: It is related to data distribution. Certain data files can reproduce the problem stably. 

SQL：
```
select id from hive.test.loan_tab where user_id = 512797;
```

Exception：
```
Query 20231128_081935_00000_r57pu failed: Failed to read ORC file: hdfs://R2/user/hive/warehouse/insurance.db/loan_tab/grass_date=2023-03-20/part-00501-0942f8ec-c6ea-444b-9c4b-0e8d4ea31cbc.c000.snappy.orc
io.trino.spi.TrinoException: Failed to read ORC file: hdfs://R2/user/hive/warehouse/test.db/loan_tab/grass_date=2023-03-20/part-00501-0942f8ec-c6ea-444b-9c4b-0e8d4ea31cbc.c000.snappy.orc
        at io.trino.plugin.hive.orc.OrcPageSource.handleException(OrcPageSource.java:201)
        at io.trino.plugin.hive.orc.OrcPageSource.getNextPage(OrcPageSource.java:155)
        at io.trino.plugin.hive.HivePageSource.getNextPage(HivePageSource.java:159)
        at io.trino.operator.ScanFilterAndProjectOperator$ConnectorPageSourceToPages.process(ScanFilterAndProjectOperator.java:379)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
        at io.trino.operator.WorkProcessorUtils$YieldingProcess.process(WorkProcessorUtils.java:181)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
        at io.trino.operator.WorkProcessorUtils$BlockingProcess.process(WorkProcessorUtils.java:207)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils.lambda$flatten$6(WorkProcessorUtils.java:317)
        at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:359)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
        at io.trino.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:240)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
        at io.trino.operator.WorkProcessorUtils.lambda$finishWhen$3(WorkProcessorUtils.java:255)
        at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
        at io.trino.operator.WorkProcessorSourceOperatorAdapter.getOutput(WorkProcessorSourceOperatorAdapter.java:132)
        at io.trino.operator.Driver.processInternal(Driver.java:395)
        at io.trino.operator.Driver.lambda$process$8(Driver.java:298)
        at io.trino.operator.Driver.tryWithLock(Driver.java:694)
        at io.trino.operator.Driver.process(Driver.java:290)
        at io.trino.operator.Driver.processForDuration(Driver.java:261)
        at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:887)
        at io.trino.execution.executor.timesharing.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:187)
        at io.trino.execution.executor.timesharing.TimeSharingTaskExecutor$TaskRunner.run(TimeSharingTaskExecutor.java:565)
        at io.trino.$gen.Trino_dev____20231128_081908_2.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 835 out of bounds for length 484
        at com.google.common.collect.RegularImmutableList.get(RegularImmutableList.java:77)
        at io.trino.orc.checkpoint.Checkpoints.getStreamCheckpoints(Checkpoints.java:75)
        at io.trino.orc.StripeReader.createRowGroups(StripeReader.java:351)
        at io.trino.orc.StripeReader.readStripe(StripeReader.java:173)
        at io.trino.orc.OrcRecordReader.advanceToNextStripe(OrcRecordReader.java:557)
        at io.trino.orc.OrcRecordReader.advanceToNextRowGroup(OrcRecordReader.java:499)
        at io.trino.orc.OrcRecordReader.nextPage(OrcRecordReader.java:420)
        at io.trino.plugin.hive.orc.OrcPageSource.getNextPage(OrcPageSource.java:150)
        ... 39 more

```

**Comment by usberkeley** [comment]
I also submitted a PR to solve the problem. #19935

**Comment by usberkeley** [comment]
I also submitted a PR to solve the problem. #19935 
cc: @martint @dain

**Comment by usberkeley** [comment]
I discovered this problem and submitted a PR, and the bug was solved in #19935

**Comment by Praveen2112** [comment]
The issue is fixed here - https://github.com/trinodb/trino/pull/19935
